go/cmd/zk/zkcmd.go

```diff
@@ -133,35 +130,41 @@ func init() {
 	}
 }
 
-var (
-	server = flag.String("server", "", "server(s) to connect to")
-)
-
 func main() {
 	defer exit.Recover()
 	defer logutil.Flush()
+	server = pflag.String("server", "", "server(s) to connect to")
+
+	// handling case of --help, -help & -h
+	args := os.Args[1:]
+	var help bool
+	if args[0] == "-h" {
+		pflag.BoolVarP(&help, "help", "h", false, "display usage and exit")
+	} else if args[0] == "--help" || args[0] == "-help" {
+		pflag.BoolVar(&help, "help", false, "display usage and exit")
+	}
```

Comment:
why not just define the `BoolVarP` regardless? (also `--help` should work even if it's not the first arg ...)

Reply:
BooVarP is only added when client is passing 'h'. If we add BoolVarP everytime then we will run into error when client pass -help. 
unknown shorthand flag: 'e' in -elp

I saw @GuptaManan100  also ran into same problem and it was fixed similar way. 
https://github.com/vitessio/vitess/pull/10911/files#diff-9a94072feabe23b4b635a733c84aa97dcf1a023e9ebc0d3a44daaf869038c8e0R109

Reply:
no, that was done because `vtorc` hadn't been migrated with the deprecation notice before the last version.

you don't need to handle that case

Reply:
(for the record, i was the one that introduced that code, not manan 😂 )

Reply:
But I am getting error if I just do BoolVarP for all cases. Haven't got time to debug through it though.

Reply:
with `pflag` you cannot pass `-help`, only `--help` and `-h` are allowed